packages/nimble-components/src/tabs/specs/scrollable-tabs.md

```diff
@@ -0,0 +1,58 @@
+
+### Out of scope of this HLD
+
+Any decision/API around limiting the width of a particular tab. This is ultimately an orthogonal concern to the UX this HLD is presenting.
+
+## Links To Relevant Work Items and Reference Material
+
+-   [Issue 1509: Scrollable Tabs](https://github.com/ni/nimble/issues/1509)
+-   [Figma design](https://www.figma.com/design/PO9mFOu5BCl8aJvFchEeuN/Nimble_Components?node-id=11133-57220&node-type=instance&t=DlnrdRo7KWYlheFz-0)
+
+## Implementation / Design
+
+![Scrollable Tabs](./spec-images/scrollable-tabs.png)
+
+The following decisions were agreed to in the comments section in the linked issue above:
+
+-   No scrollbar will be shown
+-   Keyboard navigation will work as it does today, but when a tab that isn't fully in view gets focus it should be scrolled into view
+-   When a scroll button is pressed we will scroll either a full "page width" or the remainder of the available space, and tab focus is not changed.
+
+Additionally, devices like mouse wheels and trackpads as well as pan gestures on mobile devices should allow scrolling through the tabs. This should be handled natively by the browser by simply applying CSS of `overflow: scroll;` to the div containing the tabs. _Note: Mouse users would have to hold `<Shift>` while using the scroll wheel in order to scroll the tabs. This is consistent with other horizontally scrollable content._
+
+To accomplish this we will fork FAST's template in order to add the scroll buttons. A `ResizeObserver` will be used to track when the scrollable area exceeds the viewable area in order to show/hide the scroll buttons.
+
+There will be no new API to enable this behavior, and will simply be on by default.
+
+#### TabsToolbar
+
+Only the tabs themselves will be scrollable. The scroll buttons will only surround the tabs. If there is content being displayed in a `TabsToolbar` that has been slotted, and the tabs content is wide enough to merit showing the scroll buttons, the right scroll button will sit to the left of the `TabsToolbar` (example below).
+
+![Scrollable Tabs with Toolbar](./spec-images/scrollable-tabs-toolbar.png)
+
+#### Minimum scroll area width
+
+As the tab list view width is now resizable, we must consider the scenario where the view is shrunk considerably. Preventing the scrollable area from disappearing completely seems desirable, and so we will set the CSS `min-width` property on the container of the tabs to be `4 * ${controlHeight}` (wide enough for both scroll buttons and extra space to see a good portion of at least one tab).
```

Comment:
May also be too much of an implementation detail but setting the width to be a function of control height seems wrong for two reasons:
1. We should avoid having sizing calculation be a function of control height. control height should be a default value for native css sizing and layout, i.e. height, etc. See point 2 in https://github.com/ni/nimble/issues/610#issue-1277404188
2. control height seems to be used as a size ramp token in this case. The context is not about the height of controls. In that case we could be using or adding to the size ramp tokens: https://nimble.ni.dev/storybook/?path=/docs/tokens-size-ramp--docs Those are called "padding" but are just fixed sizes without actual contextual meaning. In this case we might as well just have a fixed value since it is kinda arbitrary.

Reply:
Limited the description to simply saying we will need to set the `min-width` on the `Tab` component, and that it should be of a sufficient size to accommodate both scroll buttons.